library/std/src/sys/unix/thread.rs

```diff
@@ -132,8 +132,11 @@ impl Thread {
 
     #[cfg(target_os = "linux")]
     pub fn set_name(name: &CStr) {
+        const TASK_COMM_LEN: usize = 16;
+
         unsafe {
             // Available since glibc 2.12, musl 1.1.16, and uClibc 1.0.20.
+            let name = truncate_cstr(name, TASK_COMM_LEN);
             libc::pthread_setname_np(libc::pthread_self(), name.as_ptr());
```

Comment:
Would it make sense to at least debug_assert that no error is being returned here? Or even to assert that?

Reply:
I would be fine with adding a `debug_assert`. I don't think it's worth a full `assert` though -- it could be blocked by seccomp or something, for whatever reason, but this isn't really a critical panic-worthy failure.